modules/tags.js

```diff
@@ -0,0 +1,235 @@
+// @ts-check
+
+const Augur = require("augurbot-ts"),
+  Discord = require('discord.js'),
+  fs = require('fs'),
+  axios = require('axios'),
+  u = require("../utils/utils");
+
+/**
+ * @type {Discord.Collection<string, {tag: string, response?: string, attachment?: string, _id: import("mongoose").Types.ObjectId}>}
+ */
+let tags = new u.Collection();
+
+/**
+ * @param {string | undefined} tag The tag name to find
+ */
+const findTag = (tag) => tag ? tags.find(t => t.tag.toLowerCase() == tag.toLowerCase()) ?? null : null;
+
+/**
+ * @param {Discord.Attachment} attachment
+ * @param {{_id: import("mongoose").Types.ObjectId}} cmd
+*/
+async function saveAttachment(attachment, cmd) {
+  // @ts-ignore axios hates correct types
+  const response = await axios({
+    method: "get",
+    url: attachment.url,
+    responseType: "stream",
+  });
+  response.data.pipe(fs.createWriteStream(process.cwd() + "/media/tags/" + cmd._id.toString()));
+}
+
+/** @param {Discord.Message} msg */
+function runTag(msg) {
+  const cmd = u.parse(msg);
+  const tag = findTag(cmd?.command);
+
+  if (!tag) return;
+  let response = tag.response;
+  let target = msg.mentions?.members?.first() || msg.mentions?.users?.first();
+
+  if (response) {
+    const randomChannels = msg.inGuild() ? msg.guild.channels.cache.filter(c =>
+      c.isTextBased() && !c.isThread() &&
+      !c.permissionOverwrites?.cache.get(msg.guild.id)?.deny?.has("ViewChannel")
+    ).map(c => c.toString()) : ["Here"];
+
+    const regex = /<@random ?\[(.*?)\]>/gm;
+    if (regex.test(response)) {
+      console.log("true");
+      const replace = (/** @type {string} */ str) => u.rand(str.replace(regex, '$1').split('|'));
+      console.log(response.replace(regex, replace));
+      response = response.replace(regex, replace);
+    }
+    response = response
+      .replace(/<@author>/ig, msg.author.toString())
+      .replace(/<@authorname>/ig, msg.member?.displayName || msg.author.username)
+      .replace(/<@channel>/ig, msg.channel.toString())
+      .replace(/<@randomchannel>/, u.rand(randomChannels) ?? msg.channel.toString());
+    if ((/(<@target>)|(<@targetname>)/ig).test(response)) {
+      if (!msg.guild) target ??= msg.client.user;
+      if (!target) return msg.reply("You need to `@mention` a user with that command!").then(u.clean);
+      response = response.replace(/<@target>/ig, target.toString())
+        .replace(/<@targetname>/ig, target.displayName);
```

Comment:
If you think it's a good idea, I would recommend some way to add a newline to a tag. Since Discord doesn't seem to allow newlines in their slash commands.